paimon-core/src/main/java/org/apache/paimon/rest/RESTTokenFileIO.java

```diff
@@ -153,7 +153,8 @@ private FileIO fileIO() throws IOException {
 
             CatalogContext context = catalogLoader.context();
             Options options = context.options();
-            options = new Options(RESTUtil.merge(options.toMap(), token.token()));
+            // the original options are not overwritten
+            options = new Options(RESTUtil.merge(token.token(), options.toMap()));
```

Comment:
Is it possible to provide a separate method to get options from options and token, and add a test to protect this logic?

Reply:
> Is it possible to provide a separate method to get options from options and token, and add a test to protect this logic?

good idea! done.